ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -623,27 +630,46 @@ private static void maybeCreateProcessingLogStream(
       final KsqlEngine ksqlEngine,
       final CommandQueue commandQueue
   ) {
+    final Set<String> sourceNames = ksqlEngine.getMetaStore().getAllDataSources()
+        .keySet().stream().map(SourceName::name)
+        .collect(Collectors.toSet());
+
     if (!config.getBoolean(ProcessingLogConfig.STREAM_AUTO_CREATE)
-        || !commandQueue.isEmpty()) {
```

Comment:
This check doesn't work anymore because there could be an aborted processing log statement in the command topic.

Reply:
reverted.